rust/connlib/tunnel/src/peer.rs

```diff
@@ -315,49 +331,24 @@ impl ClientOnGateway {
         resource: connlib_shared::messages::gateway::ResourceDescription,
         expires_at: Option<DateTime<Utc>>,
     ) {
-        self.resources.insert(
-            resource.id(),
-            match resource {
-                ResourceDescription::Dns(r) => ResourceOnGateway::Dns {
-                    domains: HashMap::default(),
-                    filters: r.filters,
-                    address: r.address,
-                    expires_at,
-                },
-                ResourceDescription::Cidr(r) => ResourceOnGateway::Cidr {
-                    network: r.address,
-                    filters: r.filters,
-                    expires_at,
-                },
-                ResourceDescription::Internet(_) => ResourceOnGateway::Internet { expires_at },
-            },
-        );
```

Comment:
This was the bug @conectado. Allowing a DNS resources again overwrote the `domains` HashMap which meant that we lost all previously allowed IPs.